2022-10-02 14:43:58.862445: I tensorflow/compiler/xla/service/service.cc:170] XLA service 0x562ead1f4490 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2022-10-02 14:43:58.862967: I tensorflow/compiler/xla/service/service.cc:178]   StreamExecutor device (0): Host, Default Version
2022-10-02 14:43:59.134328: I tensorflow/compiler/xla/service/service.cc:170] XLA service 0x562ea8d68340 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2022-10-02 14:43:59.134384: I tensorflow/compiler/xla/service/service.cc:178]   StreamExecutor device (0): Quadro P6000, Compute Capability 6.1
2022-10-02 14:43:59.890022: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1532] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 23676 MB memory:  -> device: 0, name: Quadro P6000, pci bus id: 0000:04:00.0, compute capability: 6.1
*** Saving checkpoints to: /tuba/datasets/enformer/checkpoints/mcSeq2RNA_mse_2stems_20221002-144349-0
**** Running on ssrde-c-401.ucsd.edu
/home/emukamel/enformer/prepare_training_data/Li2021_mouse_snATAC_binsize128_BasenjiBins
Traceback (most recent call last):
  File "/tuba/datasets/enformer/scripts/trainnet_enformer.py", line 136, in <module>
    datasets[t] = tf.data.Dataset.zip((predictors_datasets,targets_dataset)).map(lambda x,y:{'predictors':x,'target':np.log10(1+y)})
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/data/ops/dataset_ops.py", line 2048, in map
    return MapDataset(self, map_func, preserve_cardinality=True, name=name)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/data/ops/dataset_ops.py", line 5243, in __init__
    self._map_func = structured_function.StructuredFunctionWrapper(
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/data/ops/structured_function.py", line 271, in __init__
    self._function = fn_factory()
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/eager/function.py", line 2567, in get_concrete_function
    graph_function = self._get_concrete_function_garbage_collected(
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/eager/function.py", line 2533, in _get_concrete_function_garbage_collected
    graph_function, _ = self._maybe_define_function(args, kwargs)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/eager/function.py", line 2711, in _maybe_define_function
    graph_function = self._create_graph_function(args, kwargs)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/eager/function.py", line 2627, in _create_graph_function
    func_graph_module.func_graph_from_py_func(
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/framework/func_graph.py", line 1141, in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/data/ops/structured_function.py", line 248, in wrapped_fn
    ret = wrapper_helper(*args)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/data/ops/structured_function.py", line 177, in wrapper_helper
    ret = autograph.tf_convert(self._func, ag_ctx)(*nested_args)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/autograph/impl/api.py", line 692, in wrapper
    raise e.ag_error_metadata.to_exception(e)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/autograph/impl/api.py", line 689, in wrapper
    return converted_call(f, args, kwargs, options=options)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/autograph/impl/api.py", line 439, in converted_call
    result = converted_f(*effective_args, **kwargs)
  File "/tmp/__autograph_generated_filecip5tek3.py", line 5, in <lambda>
    tf__lam = lambda x, y: ag__.with_function_scope(lambda lscope: {'predictors': x, 'target': ag__.converted_call(np.log10, (1 + y,), None, lscope)}, 'lscope', ag__.STD)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/autograph/core/function_wrappers.py", line 113, in with_function_scope
    return thunk(scope)
  File "/tmp/__autograph_generated_filecip5tek3.py", line 5, in <lambda>
    tf__lam = lambda x, y: ag__.with_function_scope(lambda lscope: {'predictors': x, 'target': ag__.converted_call(np.log10, (1 + y,), None, lscope)}, 'lscope', ag__.STD)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/autograph/impl/api.py", line 417, in converted_call
    return _call_unconverted(f, args, kwargs, options)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/autograph/impl/api.py", line 459, in _call_unconverted
    return f(*args)
  File "/home/emukamel/miniconda3/envs/tf/lib/python3.9/site-packages/tensorflow/python/framework/ops.py", line 926, in __array__
    raise NotImplementedError(
NotImplementedError: in user code:

    File "/tuba/datasets/enformer/scripts/trainnet_enformer.py", line 136, in None  *
        lambda x,y:{'predictors':x,'target':np.log10(1+y)}

    NotImplementedError: Cannot convert a symbolic tf.Tensor (add:0) to a numpy array. This error may indicate that you're trying to pass a Tensor to a NumPy call, which is not supported.

